packages/runtime/test-runtime-utils/src/test/mockLoggerErrorType.spec.ts

```diff
@@ -0,0 +1,49 @@
+/*!
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { strict as assert } from "assert";
+import {
+    ContainerErrorType,
+    IGenericError,
+    // ICriticalContainerError,
+    // IErrorBase,
+} from "@fluidframework/container-definitions";
+import { MockLogger } from "../mockLogger";
+
+ /**
+  * A set of tests to make sure that the value of "errorType" field is not lost in telemetry through all the various
+  * ways the FluidFramework codebase creates and sends errors.
+  *
+  * As of 0.21, there is no single ErrorType enum - it has been broken into several enums that each may also have
+  * sibling types. This test suite aims to be exhaustive.
+  *
+  * TODO: Consider breaking apart into multiple test files?
+  * TODO: Some tests below are actually of type aliases - kept here for completeness and in case those types become
+  * extensions or otherwise not-just-aliases in the future.
+  */
+describe("Check if the errorType field matches after sending/receiving via telemetry. Breakdown by area.", () => {
+    describe("Container errors", () => {
+        let mockLogger: MockLogger;
+        beforeEach(() => {
+            mockLogger = new MockLogger();
+        });
+
+        it("Send and receive a genericError", () => {
+            const testError: IGenericError = {
```

Comment:
this is a good start. we should keep this test and add a similar test that uses the GenericError class, and a test that uses a child logger built over the mock logger

Reply:
looks like the generice error class live in a different package. so in this file we can probably only add the child logger test. the test for the GenericError class should live in packages\loader\container-utils\src\test. to get access to the mock logger there you'll need to add a dev dependency to "@fluidframework/test-runtime-utils" in the container-utils package

Reply:
if you have any problems getting dependecies setup let me know

Reply:
actually. let's wait on that, and add the GenericTest in a follow up pr